fast64_internal/render_settings.py

```diff
@@ -0,0 +1,160 @@
+    renderSettings: "Fast64RenderSettings_Properties" = context.scene.fast64.renderSettings
+    if renderSettings.sm64Area and renderSettings.useObjectRenderPreview:
+        area: bpy.types.Object = renderSettings.sm64Area
+        renderSettings.fogPreviewColor = tuple(c for c in area.area_fog_color)
+        renderSettings.fogPreviewPosition = tuple(round(p) for p in area.area_fog_position)
+
+        renderSettings.clippingPlanes = tuple(float(p) for p in area.clipPlanes)
+
+
+def on_update_oot_render_settings(self, context: bpy.types.Context):
+    # TODO: Update render properties from selected OOTLightProperty
+    pass
+
+
+def update_lighting_space(renderSettings: "Fast64RenderSettings_Properties"):
+    if renderSettings.useWorldSpaceLighting:
+        bpy.data.node_groups["ShdCol_L"].nodes["GeometryNormal"].node_tree = bpy.data.node_groups[
+            "GeometryNormal_WorldSpace"
+        ]
+    else:
+        bpy.data.node_groups["ShdCol_L"].nodes["GeometryNormal"].node_tree = bpy.data.node_groups[
+            "GeometryNormal_ViewSpace"
+        ]
+
+
+def update_scene_props_from_render_settings(
+    context: bpy.types.Context,
+    sceneOutputs: bpy.types.NodeGroupOutput,
+    renderSettings: "Fast64RenderSettings_Properties",
+):
+    enableFog = int(renderSettings.enableFogPreview)
+    sceneOutputs.inputs["FogEnable"].default_value = enableFog
+
+    sceneOutputs.inputs["FogColor"].default_value = tuple(c for c in renderSettings.fogPreviewColor)
+    sceneOutputs.inputs["FogNear"].default_value = renderSettings.fogPreviewPosition[0]
+    sceneOutputs.inputs["FogFar"].default_value = renderSettings.fogPreviewPosition[1]
+
+    sceneOutputs.inputs["F3D_NearClip"].default_value = float(renderSettings.clippingPlanes[0])
+    sceneOutputs.inputs["F3D_FarClip"].default_value = float(renderSettings.clippingPlanes[1])
+
+    sceneOutputs.inputs["ShadeColor"].default_value = tuple(c for c in renderSettings.lightColor)
+    sceneOutputs.inputs["AmbientColor"].default_value = tuple(c for c in renderSettings.ambientColor)
+    sceneOutputs.inputs["LightDirection"].default_value = tuple(
+        d for d in (mathutils.Vector(renderSettings.lightDirection) @ transform_mtx_blender_to_n64())
+    )
+
+    update_lighting_space(renderSettings)
+
+    sceneOutputs.inputs["Blender_Game_Scale"].default_value = float(get_blender_to_game_scale(context))
+
+
+def on_update_render_preview_nodes(self, context: bpy.types.Context):
+    sceneProps = bpy.data.node_groups.get("SceneProperties")
+    if sceneProps == None:
+        print("Could not locate SceneProperties!")
+        return
+
+    sceneOutputs: bpy.types.NodeGroupOutput = sceneProps.nodes["Group Output"]
+    renderSettings: "Fast64RenderSettings_Properties" = context.scene.fast64.renderSettings
+    update_scene_props_from_render_settings(context, sceneOutputs, renderSettings)
+
+
+def on_update_render_settings(self, context: bpy.types.Context):

```

Comment:
I've noticed a couple spots where rendering code switched based on the game mode. I think it would be useful to have a brief section in the README describing the spots where one would add code if they were implementing new game specific render settings.